node/src/chain/header_validation.rs

```diff
@@ -93,6 +97,35 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
             ));
         }
 
+        if candidate_block.height != self.prev_header.height + 1 {
+            return Err(anyhow!(
+                "invalid block height block_height: {:?}, curr_height: {:?}",
+                candidate_block.height,
+                self.prev_header.height,
+            ));
+        }
```

Comment:
Is this a leftover? The same check is done right before

Reply:
Indeed. Removed